### PR TITLE
Creating Survey Modal components to facilitate survey submissions to API endpoint

### PR DESCRIPTION
**Pull request by ctrepka**
* Implements Survey Modal components to facilitate survey submissions to API endpoint related to Issue #279

### CODE REVIEW

File: src/actions/apiModalActions.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+function handleErrors(response) {
+  // Handle HTTP errors since fetch won't.
+  if (!response.ok) {
+    throw Error(response.statusText);
+  }
+  return response;
+}
+
+//-----------COMMENTED OUT FOR DEV
+// TODO: connect to new API endpoint
```
Comment (adambreznicky) [inline src/actions/apiModalActions.js:31]: is this still a TODO? pretty sure this is done

File: src/actions/index.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import * as orderCartActions from './orderCartActions';
 import * as resourceActions from './resourceActions';
 import * as toolDrawerActions from './toolDrawerActions';
 import * as urlTrackerActions from './urlTrackerActions';
+import * as apiModalActions from './apiModalActions';
```
Comment (adambreznicky) [inline src/actions/index.js:15]: in support of clean, consistent code, can we keep the import and export lists in alphabetical order? thanks

File: src/reducers/rootReducer.js [hunk 3]
```diff
@@ -32,6 +33,7 @@ const rootReducer = (history) => combineReducers({
   resources,
   toolDrawer,
   urlTracker,
+  apiModal,
```
Comment (adambreznicky) [inline src/reducers/rootReducer.js:36]: in support of clean, consistent code, can we keep the import and combineReducers lists in alphabetical order? thanks

File: src/sass/_componentSassImporter.scss [hunk 4]
```diff
@@ -25,6 +25,7 @@ them here to have them available to the app
 @import 'orderCart';
 @import 'orderCartView';
 @import 'orderTnrisDataForm';
+@import '_surveyModal';
```
Comment (adambreznicky) [inline src/sass/_componentSassImporter.scss:28]: i'm pretty certain you don't need the prefix underscore when importing your sass file. can we remove it for consistency?

**Review by adambreznicky — CHANGES_REQUESTED** [review]
i like how you utilized the standard api response "results" array and put it directly into the reducer, as-is, and as the same type without doing any manipulation or analyzing of the content within. 
i found just a few minor code standard stuff that needs cleanup before we merge this over

**Comment by ctrepka** [comment]
> i like how you utilized the standard api response "results" array and put it directly into the reducer, as-is, and as the same type without doing any manipulation or analyzing of the content within.
> i found just a few minor code standard stuff that needs cleanup before we merge this over

Thanks, just resolved those issues, should be ready to go if it all looks good

**Comment by ctrepka** [comment]
> i like how you utilized the standard api response "results" array and put it directly into the reducer, as-is, and as the same type without doing any manipulation or analyzing of the content within.

Wait was this sarcasm? Sorry, not coming thru on a screen well if it was, but is this something I should fix or is it actually desirable as a practice?

**Pull request merged**